napari/_qt/menus/debug_menu.py

```diff
@@ -0,0 +1,101 @@
+"""Debug menu.
+
+The debug menu is for developer-focused functionality that we want to be
+easy-to-use and discoverable, but which is not for the average user.
+
+"""
+from typing import TYPE_CHECKING
+
+from qtpy.QtCore import QTimer
+from qtpy.QtWidgets import QFileDialog, QMenu
+
+from ...utils import perf
+from ...utils.history import get_save_history, update_save_history
+from ...utils.translations import trans
+from ._util import populate_menu
+
+if TYPE_CHECKING:
+    from ..qt_main_window import Window
+
+
+class DebugMenu(QMenu):
+    def __init__(self, window: 'Window'):
+        self._win = window
+        super().__init__(trans._('&Debug'), window._qt_window)
+        self._perf_menu = QMenu("Performance Trace", self)
+
+        ACTIONS = [
+            {
+                'menu': self._perf_menu,
+                'items': [
+                    {
+                        'text': trans._('Start Recording...'),
+                        'slot': self._start_trace_dialog,
+                        'shortcut': 'Alt+T',
+                        'statusTip': trans._('Start recording a trace file'),
+                    },
+                    {
+                        'text': trans._('Stop Recording...'),
+                        'slot': self._stop_trace,
+                        'shortcut': 'Shift+Alt+T',
+                        'statusTip': trans._('Stop recording a trace file'),
+                    },
+                ],
+            }
+        ]
+        populate_menu(self, ACTIONS)
+        self._set_recording(False)
+        if perf.perf_config:
+            path = perf.perf_config.trace_file_on_start
+            if path is not None:
+                # Config option "trace_file_on_start" means immediately
+                # start tracing to that file. This is very useful if you
+                # want to create a trace every time you start napari,
+                # without having to start it from the debug menu.
+                self._start_trace(path)
+
+    def _start_trace_dialog(self):
```

Comment:
All these methods to not feel like they belong here

Reply:
I agree, but these were already in the [corresponding file](https://github.com/napari/napari/blob/master/napari/_qt/perf/qt_debug_menu.py) I just moved it

Reply:
I mean they can be conveniently place here, but feels odd, not a blocker ;-)

Reply:
I've got no idea how to arrange the perf stuff, so that should wait for another PR